k9mail/src/main/java/com/fsck/k9/Account.java

```diff
@@ -1177,13 +1179,8 @@ public synchronized boolean setFolderSyncMode(FolderMode syncMode) {
         FolderMode oldSyncMode = mFolderSyncMode;
         mFolderSyncMode = syncMode;
 
-        if (syncMode == FolderMode.NONE && oldSyncMode != FolderMode.NONE) {
-            return true;
-        }
-        if (syncMode != FolderMode.NONE && oldSyncMode == FolderMode.NONE) {
-            return true;
-        }
-        return false;
+        // return true iff sync mode was changed to or from FolderMode.NONE (and hence needs a refresh)
+        return (syncMode == FolderMode.NONE) != (oldSyncMode == FolderMode.NONE);
```

Comment:
I felt it was hard to read even before that and the method name wasn't very telling what the return value is, which is why I first put this in an `@return` comment in the method. Only after that I moved it, because method-header comments are the ones that get outdated after a while, which is how it ended up here.

Not attached to this though, I'll just revert :P